### PR TITLE
Updates camera docs with world units and introduces new test for intrinsics

### PR DESCRIPTION
**Pull request by pascal-roth**
# Description

Updates the camera docs with SI units and implants a new test to check that the intrinsic are set correctly for the raycaster camera.

This PR should be merged after #617 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

**Review by kellyguo11 — APPROVED** [review]

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py [hunk 1]
```diff
@@ -70,7 +70,12 @@ class GridPatternCfg(PatternBaseCfg):
 
 @configclass
 class PinholeCameraPatternCfg(PatternBaseCfg):
-    """Configuration for a pinhole camera depth image pattern for ray-casting."""
+    """Configuration for a pinhole camera depth image pattern for ray-casting.
+
+    ..note ::
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py]: Might be better to link it to the Omniverse docs page so people know where all this is coming from. We SHOULD make this a warn or caution notice so that users know that even if the doc says `mm` in USD side, it is actually a-tenth of a cm stage unit.

Might be more informative than right now.

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py [hunk 2]
```diff
@@ -81,14 +86,13 @@ class PinholeCameraPatternCfg(PatternBaseCfg):
     """
 
     horizontal_aperture: float = 20.955
-    """Horizontal aperture (in mm). Defaults to 20.955mm.
+    """Horizontal aperture (in cm). Defaults to 20.955cm.
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py]:
```suggestion
    """Horizontal aperture (in cm). Defaults to 20.955 cm.
```

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py [hunk 3]
```diff
@@ -81,14 +86,13 @@ class PinholeCameraPatternCfg(PatternBaseCfg):
...
 
     Emulates sensor/film width on a camera.
 
     Note:
         The default value is the horizontal aperture of a 35 mm spherical projector.
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py:95]: This should also be fixed?

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/ray_caster_camera_cfg.py [hunk 4]
```diff
@@ -44,6 +46,9 @@ class OffsetCfg:
     data_types: list[str] = ["distance_to_image_plane"]
     """List of sensor names/types to enable for the camera. Defaults to ["distance_to_image_plane"]."""
 
+    pattern_cfg: PinholeCameraPatternCfg = MISSING
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/ray_caster_camera_cfg.py:49]: This is just to make it enforce the type of it?

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sim/spawners/sensors/sensors_cfg.py [hunk 5]
```diff
@@ -60,14 +64,25 @@ class PinholeCameraCfg(SpawnerCfg):
     """
 
     horizontal_aperture: float = 20.955
-    """Horizontal aperture (in mm). Defaults to 20.955mm.
+    """Horizontal aperture (in cm). Defaults to 20.955cm.
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/sim/spawners/sensors/sensors_cfg.py]:
```suggestion
    """Horizontal aperture (in cm). Defaults to 20.955 cm.
```

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sim/spawners/sensors/sensors_cfg.py [hunk 6]
```diff
@@ -60,14 +64,25 @@ class PinholeCameraCfg(SpawnerCfg):
...
 
     Emulates sensor/film width on a camera.
 
     Note:
         The default value is the horizontal aperture of a 35 mm spherical projector.
     """
 
+    vertical_aperture: float | None = None
+    """Vertical aperture (in mm). Defaults to None.
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/sim/spawners/sensors/sensors_cfg.py]: Something went off with the merge here...

```suggestion
    r"""Vertical aperture (in mm). Defaults to None.
```

File: source/extensions/omni.isaac.lab/test/sensors/test_ray_caster_camera.py [hunk 7]
```diff
@@ -739,6 +739,82 @@ def test_output_equal_to_usd_camera_intrinsics(self):
             rtol=5e-6,
         )
 
+    def test_output_equal_to_usdcamera_set_intrinsics(self):
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/test/sensors/test_ray_caster_camera.py]:
```suggestion
    def test_output_equal_to_usd_camera_set_intrinsics(self):
```

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py (hunk 3 above)
Comment (pascal-roth) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py:95]: I am not sure bc I guess the 20.955 is still originating from a 35mm projector

File: source/extensions/omni.isaac.lab/test/sensors/test_ray_caster_camera.py (hunk 7 above)
Comment (Mayankm96) [reply source/extensions/omni.isaac.lab/test/sensors/test_ray_caster_camera.py]: NIT: Weird function name 😅

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/ray_caster_camera_cfg.py (hunk 4 above)
Comment (pascal-roth) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/ray_caster_camera_cfg.py:49]: yes

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py [hunk 8]
```diff
@@ -70,7 +70,13 @@ class GridPatternCfg(PatternBaseCfg):
 
 @configclass
 class PinholeCameraPatternCfg(PatternBaseCfg):
-    """Configuration for a pinhole camera depth image pattern for ray-casting."""
+    """Configuration for a pinhole camera depth image pattern for ray-casting.
+
+    .. caution::
+        Focal length as well as the aperture sizes and offsets are set as a tenth of the world unit. In our case, the
+        world unit is Meter s.t. all of these values are set in cm. For more information, please check:
```
Comment (jsmith-bdai) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py]: NIT: This was a little hard to parse

Also, seems like vertical aperature is actually in mm?
```suggestion
        world unit is meters, so all of these values are in cm. For more information, please check:
```

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py [hunk 9]
```diff
@@ -81,16 +87,15 @@ class PinholeCameraPatternCfg(PatternBaseCfg):
...
 
     Emulates sensor/film width on a camera.
 
     Note:
         The default value is the horizontal aperture of a 35 mm spherical projector.
     """
-
     vertical_aperture: float | None = None
-    """Vertical aperture (in mm). Defaults to None.
+    r"""Vertical aperture (in mm). Defaults to None.
```
Comment (jsmith-bdai) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py]: Should this be cm?
```suggestion
    r"""Vertical aperture (in cm). Defaults to None.
```

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/ray_caster_camera_cfg.py (hunk 4 above)
Comment (jsmith-bdai) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/ray_caster_camera_cfg.py:49]: NIT: This might better be named pinhole_pattern_cfg unless we expect it to be a different type of pattern cfg

**Review by jsmith-bdai — APPROVED** [review]
Looks good besides the one questions on mm->cm

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/ray_caster_camera_cfg.py (hunk 4 above)
Comment (pascal-roth) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/ray_caster_camera_cfg.py:49]: `pattern_cfg` is the general term inherited from `RayCasterCfg`, if we rename it now, that would create an ambiguity

**Review by Dhoeller19 — APPROVED** [review]
